### PR TITLE
feat：Add synchronized Alibaba cloud SMS template

### PR DESCRIPTION
**Pull request by wzh425**

### CODE REVIEW

File: src/Contracts/Masa.Mc.Contracts.Admin/Options/Channels/WebsiteMessageChannelOptions.cs [hunk 1]
```diff
@@ -2,5 +2,4 @@
 
 public class WebsiteMessageChannelOptions
 {
-    public string Environment { get; set; } = string.Empty;
```
Comment (MayueCif) [inline src/Contracts/Masa.Mc.Contracts.Admin/Options/Channels/WebsiteMessageChannelOptions.cs:5]: empty class?

File: src/Contracts/Masa.Mc.Contracts.Admin/Dtos/MessageTemplates/MessageTemplateCreateUpdateDto.cs [hunk 2]
```diff
@@ -7,13 +9,15 @@ public MessageTemplateCreateUpdateDto()
...
     public Guid ChannelId { get; set; }
+    public ChannelType ChannelType { get; set; }
     public string DisplayName { get; set; } = string.Empty;
     public string Title { get; set; } = string.Empty;
     public string Content { get; set; } = string.Empty;
     public string Example { get; set; } = string.Empty;
     public string TemplateId { get; set; } = string.Empty;
     public bool IsJump { get; set; }
     public string JumpUrl { get; set; } = string.Empty;
+    [MaxLength(10, ErrorMessage = "Name must be less than 10 characters")]
```
Comment (MayueCif) [inline src/Contracts/Masa.Mc.Contracts.Admin/Dtos/MessageTemplates/MessageTemplateCreateUpdateDto.cs:20]: why not use fluent validator

File: src/Infrastructure/Masa.Mc.Infrastructure.Sms/Services/ISmsTemplateService.cs [hunk 3]
```diff
@@ -3,4 +3,6 @@
 public interface ISmsTemplateService
 {
     Task<SmsResponseBase> GetSmsTemplateAsync(string templateCode);
+
+    Task<SmsResponseBase> GetSmsTemplateListAsync(int pageIndex = 1, int pageSize = 50);
```
Comment (MayueCif) [inline src/Infrastructure/Masa.Mc.Infrastructure.Sms/Services/ISmsTemplateService.cs:7]: rename pageindex to page

File: src/Web/Masa.Mc.Web.Admin/Pages/ReceiverGroups/Modules/ReceiverGroupCard.razor [hunk 4]
```diff
@@ -3,19 +3,20 @@
...
-       <span class="text-body3">@Model.Description</span>
+        <div class="text-body3 line-clamp-2">@Model.Description</div>
     </MCardText>
     <MCardActions class="px-4 pb-4">
-         <span>人数：@(Model.Items.Count)人</span>
-         <MSpacer/>
-         <span class="text-body2 neutral-lighten-4--text">吴自浩 编辑于 @Model.ModificationTime.Humanize() </span>
+        <span>人数：@(Model.Items.Count)人</span>
+        <MSpacer />
+        <span class="text-body2 neutral-lighten-4--text">吴自浩 编辑于 @Model.ModificationTime.Humanize() </span>
```
Comment (MayueCif) [inline src/Web/Masa.Mc.Web.Admin/Pages/ReceiverGroups/Modules/ReceiverGroupCard.razor:19]: add CultureInfo param

File: src/Web/Masa.Mc.Web.Admin/Pages/MessageTemplates/Modules/EmailTemplateCreateModal.razor.cs [hunk 5]
```diff
@@ -66,5 +66,9 @@ private void HandleVisibleChanged(bool val)
     private async Task HandleSelectChannelTypeAsync(ChannelType Type)
     {
         _channelItems = await ChannelService.GetListByTypeAsync(Type);
+        if (_channelItems.Count == 1)
```
Comment (MayueCif) [inline src/Web/Masa.Mc.Web.Admin/Pages/MessageTemplates/Modules/EmailTemplateCreateModal.razor.cs:69]: any()

File: src/Web/Masa.Mc.Web.Admin/Pages/MessageTemplates/Modules/SmsTemplateCreateModal.razor.cs [hunk 6]
```diff
@@ -66,33 +70,46 @@ private void HandleVisibleChanged(bool val)
...
     }
 
-    private void HandleChannelChange()
+    private async Task HandleChannelChangeAsync()
     {
         _model.DisplayName = string.Empty;
         _model.Content = string.Empty;
         _model.TemplateId = string.Empty;
         _model.Items = new();
+
```
Comment (MayueCif) [inline src/Web/Masa.Mc.Web.Admin/Pages/MessageTemplates/Modules/SmsTemplateCreateModal.razor.cs:96]: blank line

File: src/Web/Masa.Mc.Web.Admin/Pages/MessageTemplates/Modules/EmailTemplateCreateModal.razor.cs (hunk 5 above)
Comment (wzh425) [reply src/Web/Masa.Mc.Web.Admin/Pages/MessageTemplates/Modules/EmailTemplateCreateModal.razor.cs:69]: The requirement is that only one piece of data will be selected by default

File: src/Infrastructure/Masa.Mc.Infrastructure.Sms.Aliyun/Services/SmsTemplateService.cs [hunk 7]
```diff
@@ -22,10 +22,10 @@ public async Task<SmsResponseBase> GetSmsTemplateAsync(string templateCode)
         return new SmsTemplateResponse(body.Code == "OK", body.Message, response);
     }
 
-    public async Task<SmsResponseBase> GetSmsTemplateListAsync(int pageIndex = 1, int pageSize = 50)
+    public async Task<SmsResponseBase> GetSmsTemplateListAsync(int page = 1, int pageSize = 50)
     {
         var client = await CreateClientAsync();
-        QuerySmsTemplateListRequest querySmsTemplateListRequest = new QuerySmsTemplateListRequest() { PageIndex = pageIndex, PageSize = pageSize };
+        QuerySmsTemplateListRequest querySmsTemplateListRequest = new QuerySmsTemplateListRequest() { PageIndex = page, PageSize = pageSize };
```
Comment (MayueCif) [inline src/Infrastructure/Masa.Mc.Infrastructure.Sms.Aliyun/Services/SmsTemplateService.cs:28]: pageindex = page ->page=page

**Review by MayueCif — APPROVED** [review]

**Comment by MayueCif** [comment]
@doddgu LGTM

File: src/Infrastructure/Masa.Mc.Infrastructure.Common/Extensions/FluentValidationExtensions.cs [hunk 8]
```diff
@@ -0,0 +1,79 @@
+﻿namespace Masa.Mc.Infrastructure.Common.Extensions;
+
+public static class FluentValidationExtensions
```
Comment (doddgu) [inline src/Infrastructure/Masa.Mc.Infrastructure.Common/Extensions/FluentValidationExtensions.cs:3]: You can move the extensions to MASA.Utils.

File: src/Web/Masa.Mc.Web.Admin/Pages/MessageTemplates/EmailTemplateManagement.razor [hunk 9]
```diff
@@ -10,10 +10,10 @@
                 <MIcon Size=16 Class="mr-2 neutral-lighten-1--text">mdi-magnify</MIcon>
             </PrependInnerContent>
         </MTextField>
-        <MIcon Class="ml-4" OnClick="ToggleAdvanced">mdi-blinds</MIcon>
+        <MIcon Class="ml-4" OnClick="ToggleAdvanced">mdi-filter</MIcon>
         <MButton Color="primary" Class="ml-6 medium-button" OnClick="()=> _createModal.OpenModalAsync(ChannelType.Email)">
             <MIcon Size=18>mdi-plus</MIcon>
-            <span class="ml-2">@T("Add")</span>
+            <span class="ml-2">@T("Permission:Create")</span>
```
Comment (doddgu) [inline src/Web/Masa.Mc.Web.Admin/Pages/MessageTemplates/EmailTemplateManagement.razor:16]: Usually use `.` as a separator.

File: src/Web/Masa.Mc.Web.Admin/wwwroot/i18n/en-US.json [hunk 10]
```diff
@@ -76,12 +76,20 @@
...
   "Description:MessageTemplate.ChannelId.Required": "Please select a channel",
   "Description:MessageTemplate.TemplateId.Required": "Please select templateId",
+  "Description:MessageTemplate.Sign.Required": "Please enter your signature",
   "Description:MessageTemplate.DayLimit.Tips": "This module can send the same user at most every day",
   "DisplayName:MessageTemplateItem": "Template variable",
   "DisplayName:MessageTemplateItemCode": "Variable",
   "DisplayName:MessageTemplateItemMappingCode": "Mapping variable",
   "DisplayName:MessageTemplateItemDisplayText": "Remarks",
 
+  "DisplayName:SmsTemplateType.VerificationCode": "Verification code",
```
Comment (doddgu) [inline src/Web/Masa.Mc.Web.Admin/wwwroot/i18n/en-US.json:86]: The root don't need another seperator.

**Review by doddgu — CHANGES_REQUESTED** [review]

**Review by doddgu — APPROVED** [review]

**Pull request merged**